packages/test-runner-commands/src/sendKeysPlugin.ts

```diff
@@ -2,31 +2,59 @@ import { TestRunnerPlugin } from '@web/test-runner-core';
 import type { ChromeLauncher, puppeteerCore } from '@web/test-runner-chrome';
 import type { PlaywrightLauncher } from '@web/test-runner-playwright';
 
+type TypePayload = { type: string; };
+type PressPayload = { press: string; };
+type DownPayload = { down: string; };
+type UpPayload = { up: string; };
+
 export type SendKeysPayload =
-  | { type: string; press?: undefined }
-  | { type?: undefined; press: string };
+  | TypePayload
+  | PressPayload
+  | DownPayload
+  | UpPayload;
 
 function isObject(payload: unknown): payload is Record<string, unknown> {
   return payload != null && typeof payload === 'object';
 }
 
 function isSendKeysPayload(payload: unknown): boolean {
+  const validOptions = ['type', 'press', 'down', 'up'];
```

Comment:
Had to change the validation logic quite a bit here. Unfortunately there are no existing tests for it, so I did not add tests either. I can add them if necessary.